src/common/meta/src/ddl.rs

```diff
@@ -102,6 +102,31 @@ pub struct TableMetadata {
     pub region_wal_options: HashMap<RegionNumber, String>,
 }
 
+pub type RegionFailureDetectorControllerRef = Arc<dyn RegionFailureDetectorController>;
+
+/// Used for actively registering Region failure detectors.
+///
+/// Ensuring the Region Supervisor can detect Region failures without relying on the first heartbeat from the datanode.
+#[async_trait::async_trait]
+pub trait RegionFailureDetectorController: Send + Sync {
```

Comment:
@coderabbitai Is there a better name to replace the `RegionFailureDetectorController`